packages/flutter/lib/src/widgets/snapshot_widget.dart

```diff
@@ -0,0 +1,269 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'media_query.dart';
+
+/// Controls how the [SnapshotWidget] paints its child via the [SnapshotWidgetController].
+enum SnapshotMode {
+  /// the child is snapshotted, but only if all descendants can be snapshotted.
+  ///
+  /// If there is a platform view in the children of a raster widget, the
+  /// snapshot will not be used and the child will be rendered as normal.
```

Comment:
normal as an `SnapshotMode.normal` (which seems to indicate that this would throw) or is this a different normal?

Reply:
hmm, no - not sure how else to phrase this - the normal way as in ... super.paint

Reply:
instead of calling it normal maybe say things will be rendered like they would if no SnapshotWidget were to be used?

Reply:
but we still use the painter